WordPress/Classes/Utility/Media/MediaVideoExporter.swift

```diff
@@ -52,21 +52,30 @@ class MediaVideoExporter: MediaExporter {
     ///
     func exportVideo(with session: AVAssetExportSession, onCompletion: @escaping (MediaVideoExport) -> Void, onError: @escaping (MediaExportError) -> Void) {
         do {
-            let supportedTypes = session.supportedFileTypes
-            // Check that the exportFileType is supported.
-            // Otherwise the session will raise an exception when setting outputFileType.
-            guard supportedTypes.contains(exportFileType) else {
-                throw VideoExportError.videoExportSessionDoesNotSupportVideoOutputType
+            var outputType = preferredExportFileType
+            // Check if the exportFileType is one of the supported types for the exportSession.
+            if session.supportedFileTypes.contains(outputType) == false {
+                /* 
+                 If it is not supported by the session, try and find one
+                 of the exporter's own supported types within the session's.
+                */
+                let availableTypes = session.supportedFileTypes.filter { supportedExportFileTypes.contains($0) }
+                guard let supportedType = availableTypes.first else {
```

Comment:
Oh that's much nicer! I'll use this in an upcoming PR as I have just touched this line, again.